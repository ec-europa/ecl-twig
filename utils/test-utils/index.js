const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
