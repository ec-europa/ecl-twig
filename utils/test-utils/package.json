{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
