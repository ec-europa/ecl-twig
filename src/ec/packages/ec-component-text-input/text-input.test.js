import path from 'path';
import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

describe('EC - Text input', () => {
  const template = path.resolve(__dirname, './text-input.html.twig');
  const render = params => renderTwigFileAsNode(template, params);

  describe('Default', () => {
    const options = {
      label: 'Label',
      placeholder: 'Placeholder',
      helper_text: 'Help message',
      id: 'example-id',
      name: 'example-name',
    };

    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(options)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(options, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(options, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });
  });

  describe('Disabled', () => {
    const options = {
      label: 'Label',
      placeholder: 'Placeholder',
      helper_text: 'Help message',
      disabled: true,
      id: 'example-id',
    };

    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(options)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(options, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(options, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });
  });

  describe('With error', () => {
    const options = {
      label: 'Label',
      placeholder: 'Placeholder',
      invalid: true,
      invalid_text: 'Error message',
      helper_text: 'Help message',
      id: 'example-id',
      name: 'example-name',
    };

    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(options)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(options, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const optionsWithExtraClasses = merge(options, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(optionsWithExtraClasses)).resolves.toMatchSnapshot();
    });
  });
});
