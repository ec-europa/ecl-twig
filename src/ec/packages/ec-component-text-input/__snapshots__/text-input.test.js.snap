// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Text input Default renders correctly 1`] = `
"<div class=\\"ecl-form-group ecl-form-group--text-input\\">
  <label class=\\"ecl-form-label\\" for=\\"example-id\\"> Label </label>

  <input
    id=\\"example-id\\"
    name=\\"example-name\\"
    type=\\"text\\"
    placeholder=\\"Placeholder\\"
    class=\\"ecl-text-input\\"
  />

  <div class=\\"ecl-help-block\\">Help message</div>
</div>
"
`;

exports[`EC - Text input Disabled renders correctly 1`] = `
"<div class=\\"ecl-form-group ecl-form-group--text-input\\">
  <label class=\\"ecl-form-label\\" for=\\"example-id\\"> Label </label>

  <input
    id=\\"example-id\\"
    name=\\"\\"
    type=\\"text\\"
    placeholder=\\"Placeholder\\"
    class=\\"ecl-text-input\\"
    disabled
  />

  <div class=\\"ecl-help-block\\">Help message</div>
</div>
"
`;

exports[`EC - Text input With error renders correctly 1`] = `
"<div class=\\"ecl-form-group ecl-form-group--text-input\\">
  <label class=\\"ecl-form-label ecl-form-label--invalid\\" for=\\"example-id\\">
    Label
  </label>

  <input
    id=\\"example-id\\"
    name=\\"example-name\\"
    type=\\"text\\"
    placeholder=\\"Placeholder\\"
    class=\\"ecl-text-input ecl-text-input--invalid\\"
  />

  <div class=\\"ecl-feedback-message\\">Error message</div>

  <div class=\\"ecl-help-block\\">Help message</div>
</div>
"
`;
